packages/client-engine-runtime/src/interpreter/QueryInterpreter.ts

```diff
@@ -2,41 +2,46 @@ import { SqlQuery, SqlQueryable } from '@prisma/driver-adapter-utils'
 
 import { QueryEvent } from '../events'
 import { JoinExpression, QueryPlanNode } from '../QueryPlan'
+import { type TransactionManager } from '../transactionManager/TransactionManager'
 import { GeneratorRegistry, GeneratorRegistrySnapshot } from './generators'
 import { renderQuery } from './renderQuery'
 import { PrismaObject, ScopeBindings, Value } from './scope'
 import { serialize } from './serialize'
 
 export type QueryInterpreterOptions = {
-  queryable: SqlQueryable
+  transactionManager: TransactionManager
+  allowTransaction: boolean
```

Comment:
```suggestion
  transactionManager: TransactionManager
  allowTransaction: boolean
```
maybe it makes sense to not pass a transaction manager at all if we don't allow transactions, I don't like having a transaction manager here when it should not be used, so it could maybe be optional

it might even be worth making it explicit via a type 
```ts
type InterpreterTransactionManager
  = { enabled: true,  manager: TransactionManager }
  | { enabled: false }
```

Reply:
Good point, let's make invalid state unrepresentable.